packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -59,6 +61,23 @@ export default class AddFragment extends BaseDialog {
         this.dialog.open();
     }
 
+    private specialIndexHandling(specialIndexAggregation: string | number): void {
+        const overlay = OverlayRegistry.getOverlay(this.runtimeControl as UI5Element);
+        const aggregations = overlay.getDesignTimeMetadata().getData().aggregations;
+
+        if(specialIndexAggregation in aggregations && 'specialIndexHandling' in aggregations[specialIndexAggregation]) {
+            this.model.setProperty('/indexHandlingFlag', false);
+            this.model.setProperty('/specialIndexHandlingIcon', true);
+            const controlType = this.runtimeControl.getMetadata().getName();
```

Comment:
This line `const controlType = this.runtimeControl.getMetadata().getName();` can be moved to the top of the current scope.

Reply:
updated in [37cc141](https://github.com/SAP/open-ux-tools/pull/1477/commits/37cc141c0f4e040454ed96e66536e4c64f27436b)